test/e2e/suite/command/manifest_index.go

```diff
@@ -519,6 +641,37 @@ var _ = Describe("OCI image layout users:", func() {
 			ValidateIndex(content, expectedManifests)
 		})
 
+		It("should update artifactType", func() {
+			root := PrepareTempOCI(ImageRepo)
+			// source index
+			ORAS("manifest", "index", "create", Flags.Layout, LayoutRef(root, "index"),
+				string(multi_arch.LinuxAMD64.Digest)).Exec()
+			// update index with artifact type
+			artifactType := "application/vnd.test-example"
+			ORAS("manifest", "index", "update", Flags.Layout, LayoutRef(root, "index"),
+				"--artifact-type", artifactType).Exec()
+			// verify
+			content := ORAS("manifest", "fetch", Flags.Layout, LayoutRef(root, "index")).Exec().Out.Contents()
+			var index ocispec.Index
+			Expect(json.Unmarshal(content, &index)).ShouldNot(HaveOccurred())
+			Expect(index.ArtifactType).To(Equal(artifactType))
+		})
+
+		It("should remove artifactType", func() {
+			root := PrepareTempOCI(ImageRepo)
+			// source index
+			ORAS("manifest", "index", "create", Flags.Layout, LayoutRef(root, "index"),
+				"--artifact-type", "application/vnd.old", string(multi_arch.LinuxAMD64.Digest)).Exec()
+			// update index with artifact type
+			ORAS("manifest", "index", "update", Flags.Layout, LayoutRef(root, "index"),
+				"--artifact-type", "").Exec()
+			// verify
+			content := ORAS("manifest", "fetch", Flags.Layout, LayoutRef(root, "index")).Exec().Out.Contents()
+			var index ocispec.Index
+			Expect(json.Unmarshal(content, &index)).ShouldNot(HaveOccurred())
+			Expect(index.ArtifactType).To(BeEmpty())
+		})
+
```

Comment:
nit: Should we also validate that the artifact type won't be removed if the `--artifact-type` flag is not specified?

Reply:
I have added an additional test below for running without any `--artifact-type` option at all, when the input already has an artifact type.